gitpod-io/spring-petclinic
# Leave only the required files in the .idea folder and update the Run Configuration to use Maven Wrapper

## Description

We want to be confident about which files from the .idea folder are really necessary to make a Maven Project completely indexed during the IDE Warmup (which happens during Gitpod's Prebuilds). After investigation, we concluded we only needed three files for that:
https://github.com/gitpod-io/spring-petclinic/blob/9da769cdfb80024e842c0b2e9eb3f22da27b2198/.idea/.gitignore#L5-L8

And we also wanted to demonstrate how users can share Run Configurations for JetBrains IDEs via Gitpod, so we also leave them in .idea folder:
https://github.com/gitpod-io/spring-petclinic/blob/9da769cdfb80024e842c0b2e9eb3f22da27b2198/.idea/.gitignore#L2-L4

And as we're using Maven Wrapper, we want to use [Run/Debug Configuration: Maven](https://www.jetbrains.com/help/idea/run-debug-configuration-maven.html) instead of the auto-generated configuration.

## How to test

1. Select IntelliJ IDEA in Gitpod's IDE Preferences (both Stable and EAP versions work): https://gitpod.io/preferences
2. Open this PR branch (which is already prebuilt [[1](https://gitpod.io/t/gitpod/spring-petclinic/f81a3689-356b-4d65-9ed2-6d3aea283323)]): https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic/pull/68
3. Check if the PetClinic server is started as soon as the JetBrains Client opens. Notice it reads the indexed files, but it's quick as it's already indexed.
    <img width="1369" alt="image" src="https://user-images.githubusercontent.com/418083/193799463-d22e8143-b848-48ff-9f3c-8492f58270c1.png">
4. Stop the PetClinic server on the terminal and click Play/Debug button via IntelliJ interface, and confirm it runs the server.
    <img width="1369" alt="image" src="https://user-images.githubusercontent.com/418083/193793293-843f8dc3-4eb7-431e-8a4c-1b994e5d39ba.png">
5. Check the port forward:
    a. If you're using the _EAP_ version of IntelliJ IDEA, ports are automatically forwarded, so you can check if the website is available at [http://127.0.0.1:8080](http://127.0.0.1:8080)
    <img width="1022" alt="image" src="https://user-images.githubusercontent.com/418083/193792623-96115485-48f1-446f-a459-78aa0ec11fbd.png">
    b. If you're using the _Stable_ version, you need to forward it manually:
        <img width="633" alt="image" src="https://user-images.githubusercontent.com/418083/193798442-ef037071-b588-4eb1-881e-d45182eb54d0.png">
